Corrupted bpf_features.h

```
2020-04-03T14:27:02.178302737Z level=error msg="Command execution failed" cmd="[/var/lib/cilium/bpf/init.sh /var/lib/cilium/bpf /var/run/cilium/state 10.10.1.165 fd04::12 vxlan enp0s8 <nil> <nil> 1500 false false <nil> true true /var/run/cilium/cgroupv2 /sys/fs/bpf true v2]" error="exit status 1" subsys=datapath-loader
[...]
2020-04-03T14:27:02.180517863Z level=warning msg="+ clang -O2 -target bpf -emit-llvm -Wall -Wextra -Werror -Wno-address-of-packed-member -Wno-unknown-warning-option -Wno-gnu-variable-sized-type-not-at-end -I. -I/run/cilium/state/globals -I/var/lib/cilium/bpf -I/var/lib/cilium/bpf/include -D__NR_CPUS__=2 -DENABLE_ARP_RESPONDER=1 -DHANDLE_NS=1 -DSECLABEL=2 -DLB_L3 -DLB_L4 -DDISABLE_LOOPBACK_LB -DBPF_PKT_DIR=1 '-DNODE_MAC={.addr={0x72,0x3b,0x45,0x3b,0x56,0x9d}}' -DCALLS_MAP=cilium_calls_overlay_2 -c /var/lib/cilium/bpf/bpf_overlay.c -o -" subsys=datapath-loader
2020-04-03T14:27:02.180520664Z level=warning msg="In file included from /var/lib/cilium/bpf/bpf_overlay.c:4:" subsys=datapath-loader
2020-04-03T14:27:02.180522844Z level=warning msg="In file included from /var/lib/cilium/bpf/include/bpf/ctx/skb.h:11:" subsys=datapath-loader
2020-04-03T14:27:02.180525046Z level=warning msg="In file included from /var/lib/cilium/bpf/include/linux/../bpf/helpers_skb.h:11:" subsys=datapath-loader
2020-04-03T14:27:02.180527268Z level=warning msg="In file included from /var/lib/cilium/bpf/include/linux/../bpf/features_skb.h:7:" subsys=datapath-loader
2020-04-03T14:27:02.18052949Z level=warning msg="In file included from /var/lib/cilium/bpf/include/linux/../bpf/features.h:7:" subsys=datapath-loader
2020-04-03T14:27:02.180631807Z level=warning msg="/run/cilium/state/globals/bpf_features.h:3165:60: error: ISO C99 requires whitespace after the macro name [-Werror,-Wc99-extensions]" subsys=datapath-loader
2020-04-03T14:27:02.180638469Z level=warning msg="#define BPF__PROG_TYPE_tracing__HELPER_bpf_probe_read_kerne#endif /* BPF_FEATURES_H_ */" subsys=datapath-loader
2020-04-03T14:27:02.180640895Z level=warning msg="                                                           ^" subsys=datapath-loader
2020-04-03T14:27:02.180643092Z level=warning msg="/run/cilium/state/globals/bpf_features.h:1:2: error: unterminated conditional directive" subsys=datapath-loader
2020-04-03T14:27:02.180649289Z level=warning msg="#ifndef BPF_FEATURES_H_" subsys=datapath-loader
2020-04-03T14:27:02.180652497Z level=warning msg=" ^" subsys=datapath-loader
2020-04-03T14:27:02.180654666Z level=warning msg="2 errors generated." subsys=datapath-loader
2020-04-03T14:27:02.180705683Z level=error msg="Error while initializing daemon" error="exit status 1" subsys=daemon
2020-04-03T14:27:02.18071275Z level=fatal msg="Error while creating daemon" error="exit status 1" subsys=daemon
```
Noticed in https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/18496/testReport/Suite-k8s-1/11/K8sDatapathConfig_MonitorAggregation_Checks_that_monitor_aggregation_restricts_notifications/.

[f9f3b3b1_K8sDatapathConfig_MonitorAggregation_Checks_that_monitor_aggregation_restricts_notifications.zip](https://github.com/cilium/cilium/files/4437020/f9f3b3b1_K8sDatapathConfig_MonitorAggregation_Checks_that_monitor_aggregation_restricts_notifications.zip)

This is likely happening because, although we wait for the producer command to finish, we don't wait for the consumer goroutine:
https://github.com/cilium/cilium/blob/a2441435558ddaf0da04e64577ceebe6c5a74731/pkg/datapath/linux/probes/probes.go#L252-L267

/cc @mrostecki

--- a/pkg/datapath/linux/probes/probes.go
+++ b/pkg/datapath/linux/probes/probes.go
@@ -252,7 +252,7 @@ func (p *ProbeManager) writeHeaders(featuresFile io.Writer) error {
 	io.WriteString(writer, "#ifndef BPF_FEATURES_H_\n")
 	io.WriteString(writer, "#define BPF_FEATURES_H_\n\n")
 
-	go io.Copy(writer, stdoutPipe)
+	io.Copy(writer, stdoutPipe)
 	if err := cmd.Wait(); err != nil {
 		stderr, err := ioutil.ReadAll(stderrPipe)
 		if err != nil {

--- a/test/helpers/cons.go
+++ b/test/helpers/cons.go
@@ -213,6 +213,8 @@ const (
 	emptyBPFInitArg   = "empty argument passed to bpf/init.sh"       // from https://github.com/cilium/cilium/issues/10228
 	RemovingMapMsg    = "Removing map to allow for property upgrade" // from https://github.com/cilium/cilium/pull/10626
 	logBufferMessage  = "Log buffer too small to dump verifier log"  // from https://github.com/cilium/cilium/issues/10517
+	ClangErrorsMsg    = " errors generated."                         // from https://github.com/cilium/cilium/issues/10857
+	ClangErrorMsg     = "1 error generated."                         // from https://github.com/cilium/cilium/issues/10857
 
 	// HelmTemplate is the location of the Helm templates to install Cilium
 	HelmTemplate = "../install/kubernetes/cilium"
@@ -268,6 +270,8 @@ var badLogMessages = map[string][]string{
 	emptyBPFInitArg:   nil,
 	RemovingMapMsg:    nil,
 	logBufferMessage:  nil,
+	ClangErrorsMsg:    nil,
+	ClangErrorMsg:     nil,
 }
 
 var ciliumCLICommands = map[string]string{

--- a/test/k8sT/DatapathConfiguration.go
+++ b/test/k8sT/DatapathConfiguration.go
@@ -19,6 +19,7 @@ import (
 	"io/ioutil"
 	"regexp"
 	"strconv"
+	"strings"
 
 	"github.com/cilium/cilium/test/config"
 	. "github.com/cilium/cilium/test/ginkgo-ext"
@@ -73,7 +74,12 @@ var _ = Describe("K8sDatapathConfig", func() {
 				"Service is deleted")
 		}
 
-		kubectl.ValidateNoErrorsInLogs(CurrentGinkgoTestDescription().Duration)
+		blacklist := helpers.GetBadLogMessages()
+		if strings.Contains(CurrentGinkgoTestDescription().TestText, "sockops") {
+			delete(blacklist, helpers.ClangErrorsMsg)
+			delete(blacklist, helpers.ClangErrorMsg)
+		}
+		kubectl.ValidateListOfErrorsInLogs(CurrentGinkgoTestDescription().Duration, blacklist)
 	})
 
 	deployNetperf := func() {

